libraries/azure_data_factories.rb

```diff
@@ -0,0 +1,95 @@
+    #   - resource_id => Optional parameter. If exists, `resource_group` and `name` must not be provided.
+    #     In the following format:
+    #       /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/
+    #       Microsoft.DataFactory/factories?api-version={api-version}
+    #   - api_version => Optional parameter. The latest version will be used unless provided. api-version
+    #
+    #   **`resource_group` and (resource) `name` or `resource_id`  will be validated in the backend appropriately.
+    #     We don't have to do anything here.
+    #
+    # Following resource parameters have to be defined here.
+    #   - resource_provider => Microsoft.DataFactory/factories
+    #     The `specific_resource_constraint` method will validate the user input
+    #       not to accept a different `resource_provider`.
+    #
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.DataFactory/factories', opts)
+    # static_resource parameter must be true for setting the resource_provider in the backend.
+    super(opts, true)
+    # Check if the resource is failed.
+    # It is recommended to check that after every usage of superclass methods or API calls.
+    return if failed_resource?
+
+    # Define the column and field names for FilterTable.
+    #   - column: It is defined as an instance method, callable on the resource, and present `field` values in a list.
+    #   - field: It has to be identical with the `key` names in @table items that will be presented in the FilterTable.
+    # @see https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md
+
+    table_schema = [
+      { column: :names, field: :name },
+      { column: :types, field: :type },
+      { column: :ids, field: :id },
+      { column: :tags, field: :tags },
+      { column: :provisioning_states, field: :provisioningState },
+      { column: :locations, field: :location },
+    ]
+
+    # FilterTable is populated at the very end due to being an expensive operation.
+    AzureGenericResources.populate_filter_table(:table, table_schema)
+  end
+
+  def to_s
+    super(AzureDataFactories)
+  end
+
+  private
+
+  # Populate the @table with the resource attributes.
+  # @table has been declared in the super class as an empty array.
+  # Each item in the @table
+  #   - should be a Hash object
+  #   - should have the exact key names defined in the @table_schema as `field`.
+  def populate_table
+    # If @resources empty than @table should stay as an empty array as declared in superclass.
+    # This will ensure constructing resource and passing `should_not exist` test.
+    return [] if @resources.empty?
+
+    @resources.each do |resource|
+      @table << {
+        id: resource[:id],
+        name: resource[:name],
+        type: resource[:type],
+        tags: resource[:tags],
+        provisioningState: resource[:properties][:provisioningState],
```

Comment:
why not other properties?

Reply:
Hi Sateesh 
I will add etag property apart from that others are optional so it doesn't have any impact.
Providing json response for reference . 

{
  "name": "exampleFactoryName",
  "id": "/subscriptions/12345678-1234-1234-1234-12345678abc/resourceGroups/exampleResourceGroup/providers/Microsoft.DataFactory/factories/exampleFactoryName",
  "type": "Microsoft.DataFactory/factories",
  "properties": {
    "provisioningState": "Succeeded",
    "createTime": "2018-06-19T05:41:50.0041314Z",
    "version": "2018-06-01",
    "repoConfiguration": {
      "type": "FactoryVSTSConfiguration",
      "projectName": "project",
      "tenantId": "",
      "accountName": "ADF",
      "repositoryName": "repo",
      "collaborationBranch": "master",
      "rootFolder": "/",
      "lastCommitId": ""
    }
  },
  "eTag": "\"00004004-0000-0000-0000-5b28979e0000\"",
  "location": "East US",
  "tags": {
    "exampleTag": "exampleValue"
  }
}